### PR TITLE
Subscribe to vehicle data only after HMI sends successful response 

### PR DESCRIPTION
**Pull request by VVeremjova**
Only after SDL receive successful result from HMI SDL must store store parameter

### CODE REVIEW

**Comment by VVeremjova** [comment]
@LevchenkoS, @pold500 please review

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  ~SubscribeVehicleDataRequestTest() {
+    // Fix DataAccessor release and WinQt crash
+    Mock::VerifyAndClearExpectations(&app_mngr_);
+  }
+
+ protected:
+  MockAppPtr app_;
+  //  MessageSharedPtr command_msg_;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @VVeremjova, Are this needed?

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  EXPECT_CALL(*mock_app, IsSubscribedToIVI(_)).WillRepeatedly(Return(false));
+
+  am::ApplicationSet app_set;
+  app_set.insert(mock_app);
+
+  DataAccessor<am::ApplicationSet> accessor(app_set, app_set_lock_);
+  EXPECT_CALL(app_mngr_, applications()).WillOnce(Return(accessor));
+
+  // Command was sent to HMI
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_));
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @VVeremjova, are we really, don't interested in argument?

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+  EXPECT_CALL(mock_message_helper_,
+              HMIToMobileResult(hmi_apis::Common_Result::SUCCESS))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS), _));
+  // Application will subscribe data
+  EXPECT_CALL(*mock_app, SubscribeToIVI(_));
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @VVeremjova, are we really, don't interested in argument?

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @LevchenkoS In this case - yes. This test check will app subscribe after HMI respond or not, so check here message that will be sent on HMI is not necessary

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc (hunk 3 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @LevchenkoS See 1c249fa

File: src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc [hunk 4]
```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2016, Ford Motor Company
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @VVeremjova Please update to 2017.

Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc]: @dcherniev See  c5d8e33

**Pull request merged**